modules/sfp_malwarepatrol.py

```diff
@@ -45,30 +47,44 @@ def watchedEvents(self):
     # This is to support the end user in selecting modules based on events
     # produced.
     def producedEvents(self):
-        return ["MALICIOUS_IPADDR", "MALICIOUS_INTERNET_NAME",
-                "MALICIOUS_AFFILIATE_IPADDR", "MALICIOUS_AFFILIATE_INTERNET_NAME",
-                "MALICIOUS_COHOST"]
+        return [
+            "MALICIOUS_IPADDR",
+            "MALICIOUS_INTERNET_NAME",
+            "MALICIOUS_AFFILIATE_IPADDR",
+            "MALICIOUS_AFFILIATE_INTERNET_NAME",
+            "MALICIOUS_COHOST"
+        ]
 
     def queryAddr(self, qaddr):
-        data = dict()
-        url = "http://lists.malwarepatrol.net/cgi/getfile?receipt=" + \
-              self.opts['api_key'] + "&product=8&list=smoothwall"
-
-        data['content'] = self.sf.cacheGet("sfmalwarepatrol", 72)
-        if data['content'] is None:
-            data = self.sf.fetchUrl(url, useragent=self.opts['_useragent'])
+        block_list = [
+            "clamav_basic",
+            "firekeeper",
+            "mailwasher",
+            "mozilla_adblock",
+            "spamassassin",
+            "smoothwall"
+        ]
+
+        for block_name in block_list:
+            data = dict()
+            url = "http://lists.malwarepatrol.net/cgi/getfile?receipt={}&product=32&list={}"
+            url = url.format(self.opts['api_key'], block_name)
+
+            data['content'] = self.sf.cacheGet("sfmalwarepatrol", 72)
```

Comment:
The Cache ID will need to be different for each block list, otherwise only the last in the list will be the one cached.